lib/ansible/playbook/role/__init__.py

```diff
@@ -262,11 +262,18 @@ def _load_role_data(self, role_include, parent_role=None):
 
         if self._should_validate:
             role_argspecs = self._get_role_argspecs()
-            task_data = self._prepend_validation_task(task_data, role_argspecs)
+            validation_task_data = self._prepend_validation_task(None, role_argspecs)
+            validation_blocks = load_list_of_blocks(
+                validation_task_data, play=self._play, role=self, loader=self._loader, variable_manager=self._variable_manager
+            )
+            for block in validation_blocks:
+                for task in block.block:
+                    task.implicit = True
```

Comment:
This won't change how the validation task is displayed by default. Callbacks only ignore implicit Tasks (unless they set the attr `wants_implicit_tasks`), and always get the TaskResults, even from implicit Tasks. https://github.com/ansible/ansible/blob/devel/lib/ansible/executor/task_queue_manager.py#L442-L443